cookbooks/aws-parallelcluster-config/resources/mount_fsx.rb

```diff
@@ -23,12 +23,12 @@
 default_action :mount
 
 action :mount do
-  fsx_fs_id_array = new_resource.fsx_fs_id_array
-  fsx_fs_type_array = new_resource.fsx_fs_type_array
-  fsx_shared_dir_array = new_resource.fsx_shared_dir_array
-  fsx_dns_name_array = new_resource.fsx_dns_name_array
-  fsx_mount_name_array = new_resource.fsx_mount_name_array
-  fsx_volume_junction_path_array = new_resource.fsx_volume_junction_path_array
+  fsx_fs_id_array = new_resource.fsx_fs_id_array.dup
+  fsx_fs_type_array = new_resource.fsx_fs_type_array.dup
+  fsx_shared_dir_array = new_resource.fsx_shared_dir_array.dup
+  fsx_dns_name_array = new_resource.fsx_dns_name_array.dup
+  fsx_mount_name_array = new_resource.fsx_mount_name_array.dup
+  fsx_volume_junction_path_array = new_resource.fsx_volume_junction_path_array.dup
```

Comment:
Node attributes are read-only when you do not specify which precedence level to set. We have something like `shared_dir_array[index] = "/#{shared_dir_array[index]}" unless shared_dir_array[index].start_with?('/')` in our code, which modify the value. By using dup, we are modifying the duplication of the value instead of node attribute